### ISSUE TITLE
Volatile BooleanType not possible

### ISSUE THREAD

**Issue by hanslovsky** [imglib/imglib2#209]
Both `Volatile` and the `BooleanType` interface define a `get` method. This prohibits implementation of a `VolatileBooleanType`:
```java
public abstract class VolatileBooleanType< B extends BooleanType< B >, T extends VolatileBooleanType< B, T > >
		extends Volatile< B >
		implements BooleanType< T >

{
 
	@Override
	public boolean get()
	{
		// TODO Auto-generated method stub
		return false;
	}

	// other members and methods


}
```
results in this compiler error:
```
The return type is incompatible with Volatile<B>.get()
```

This could resolved by renaming either `Volatile.get` or `BooleanType.get`. I suggest to rename `BooleanType.get` into something more meaningful, e.g. `BooleanType.isTrue`. Of course, this would be an interface breaking change. If this is the way to go we should start by deprecating `BooleanType.get` and introduce the interface change only when enough interface breaking changes have accumulated to justify a new major version.

**Comment by axtimwalde** [comment]
Isn't that the same for all other types?  https://github.com/imglib/imglib2/blob/master/src/main/java/net/imglib2/type/numeric/real/FloatType.java#L106

**Comment by hanslovsky** [comment]
No, because the `RealType` interface does not specify a `get` method (but the `BooleanType` interface does).

**Comment by axtimwalde** [comment]
Oh---not in the interface, I get it.

**Comment by axtimwalde** [comment]
OR we change `Volatile.get()` into `Volatile.value()`, probably better because `BooleanType` has been there earlier...

**Comment by hanslovsky** [comment]
`BooleanType` is probably used more than `Volatile` (in downstream code) but in my opinion, the `get` method is named more appropriately for the `Volatile` class because it is an actual wrapper around an object, whereas the `BooleanType` interface mimics a `boolean` on object level.
Ideally I would like rename both methods (`get` on its own is not very descriptive) but that might be overkill.

**Comment by ctrueden** [comment]
@hanslovsky How does this relate to #208? In that PR I (naively) added a `VolatileBooleanArray`. Would that solve your issues here?

**Comment by hanslovsky** [comment]
@ctrueden Unfortunately, this does not resolve this issue. We cannot extend `Volatile` and implement `BooleanType` because both define a method `get()`. Another solution that I just realized exists is making an `interface Volatile` instead of `class Volatile`. Then, the `get` method in the `Volatile` interface would become obsolete (and `Volatile` would only indicate `isValid`). While this seems to be the cleanest solution, it would also be the one breaking more code than any of the other proposals.

**Comment by axtimwalde** [comment]
@hanslovsky Agree, that is the nicest solution.  Not sure how much work this will be...
